### PR TITLE
WIP: Registration server

### PR DESCRIPTION
**Pull request by nkottary**
GitHub listener that detects merged pull requests on repos and calls `register`.

TODO:
* Run register only for PRs with Project file changes
* Create pull request on registry

### CODE REVIEW

File: src/Server.jl [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+        commit = payload["pull_request"]["head"]["sha"]
+    elseif kind == "status"
+        commit = payload["commit"]["sha"]
+    elseif kind == "pull_request_review"
+        commit = payload["review"]["commit_id"]
+    end
+    commit
+end
+
+event_queue = Queue{RegParams}()
```
Comment (oxinabox) [inline src/Server.jl]: Are you sure you need a actual `DataStructure.Queue` here?
`pushfirst!` and `pop!` on a `Vector` are more performant than one might expect

Comment (nkottary) [reply src/Server.jl]: Yeah, Queue is not really needed here.

Comment (oxinabox) [reply src/Server.jl]: You will thus be be able to drop the DataStructures.jl dependency

Comment (EricForgy) [reply src/Server.jl]: This all looks way over my head, but if it still needs to be done, I cold probably help remove DataStructures dependency.

Comment (EricForgy) [reply src/Server.jl]: https://github.com/JuliaComputing/Registrator.jl/pull/6

**Pull request merged**